samples/easyaccess-test.cpp

```diff
@@ -1,87 +1,86 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 // Sample program using high-level metadata access functions
 
+#include <cassert>
 #include <exiv2/exiv2.hpp>
-#include <iostream>
 #include <iomanip>
-#include <cassert>
+#include <iostream>
 
 using EasyAccessFct = Exiv2::ExifData::const_iterator (*)(const Exiv2::ExifData&);
 
-struct EasyAccess {
-    const char*   label_;
+struct EasyAccess
+{
+    const char* label_;
     EasyAccessFct findFct_;
 };
 
-static const EasyAccess easyAccess[] = {
-    { "Orientation",          Exiv2::orientation      },
-    { "ISO speed",            Exiv2::isoSpeed         },
-    { "Date & time original", Exiv2::dateTimeOriginal },
-    { "Flash bias",           Exiv2::flashBias        },
-    { "Exposure mode",        Exiv2::exposureMode     },
-    { "Scene mode",           Exiv2::sceneMode        },
-    { "Macro mode",           Exiv2::macroMode        },
-    { "Image quality",        Exiv2::imageQuality     },
-    { "White balance",        Exiv2::whiteBalance     },
-    { "Lens name",            Exiv2::lensName         },
-    { "Saturation",           Exiv2::saturation       },
-    { "Sharpness",            Exiv2::sharpness        },
-    { "Contrast",             Exiv2::contrast         },
-    { "Scene capture type",   Exiv2::sceneCaptureType },
-    { "Metering mode",        Exiv2::meteringMode     },
-    { "Camera make",          Exiv2::make             },
-    { "Camera model",         Exiv2::model            },
-    { "Exposure time",        Exiv2::exposureTime     },
-    { "FNumber",              Exiv2::fNumber          },
-    { "Shutter speed value",  Exiv2::shutterSpeedValue },
-    { "Aperture value",       Exiv2::apertureValue    },
-    { "Brightness value",     Exiv2::brightnessValue  },
-    { "Exposure bias",        Exiv2::exposureBiasValue },
-    { "Max aperture value",   Exiv2::maxApertureValue },
-    { "Subject distance",     Exiv2::subjectDistance  },
-    { "Light source",         Exiv2::lightSource      },
-    { "Flash",                Exiv2::flash            },
-    { "Camera serial number", Exiv2::serialNumber     },
-    { "Focal length",         Exiv2::focalLength      },
-    { "Subject location/area", Exiv2::subjectArea     },
-    { "Flash energy",         Exiv2::flashEnergy      },
-    { "Exposure index",       Exiv2::exposureIndex    },
-    { "Sensing method",       Exiv2::sensingMethod    },
-    { "AF point",             Exiv2::afPoint          }
-};
+static const EasyAccess easyAccess[] = {{"Orientation", Exiv2::orientation},
```

Comment:
formatting here is nasty. Can be fixed by adding a comma at the end of the last member.

Reply:
I agree on this. I invested some time trying to find a way to split the arrays in an homogeneous way (keeping always the same indentation, separating each item in a different line, etc) but it looks difficult to achieve. I was specially playing with the file `src/fujimn_int.cpp`. 

I'll keep investigating how to make the formatting of arrays more "beautiful". Suggestions here are welcome!

Reply:
In the last commit (ad7f1784a460439bd7616594154cfb2f96b8b94f) I wanted to show you what's the effect of adding a comma after the last item in the array.

In most of the cases, it makes the formatting better (see `fujiOffOn`, `fujiTone` and others) and it split each item in a different line. However, there are some cases in which clang-format decides to "pack" several items in the same line (see `fujiSharpness` or `fujiFlashMode`). That's something I would really like to change if it is possible. I would like to see a more consistent behavior for all the cases.

Reply:
You can work around it by adding // at the end of each entry.

The issue interestinly enough has to do with the max characters per line. If you remove that it won't pack them.

Reply:
I really don't like adding `//` at the end of the lines. It sets a bad example, as people will then just start using comments to hack around what `clang-format` does.

Reply:
@hassec I just had to add the `//` in few cases. If none of the items have comments at their right side, then it is not needed to add more comments. I just had to do that trick in cases like the following one, where there was a unique comment in the last item of the array: https://github.com/Exiv2/exiv2/commit/5ef2f8bd1e52e01093e8bf9d3a73cb3eb288021f#diff-8e58de4a7a00f0a4e00754c14143edab88e415b0e92b8545f977c5b1fe478681R1407. 

I agree with you that we should not start adding many of such "tricks" and expect from contributors to know them or apply them. 

On the other hand, our codebase is full of such arrays with many items inside, and I would really like to have a consistent formatting for them.

Reply:
I'm not sure if I like setting `ColumnLimit : 0` as that disables e.g. the reformatting of functions etc. 

take this example:

```cpp
// if limit is 120
int test(uint32_t LongNameForParameter1, double LongNameForParameter2, const float &LongNameForParameter3);

// if limit is 80
int test(uint32_t LongNameForParameter1, double LongNameForParameter2,
         const float &LongNameForParameter3);
```

this won't get reformatted to a common form as `clang-format` will simply use the input provided spacing if you set `ColumnLimit` to `0`

Reply:
Right, that was also a concern I had. I think that what we can see right now is Okayish because I firstly applied clang-format to the whole project with the ColumnLimit set to 120 and in the last commits (with `ColumnLimit=0`) it kept the formatting somehow. 

It would probably better to just keep the `.clang-format` with the original version. The only drawback is that some arrays are not "perfectly" formatted.

Keep your suggestions & thought flowing! Thanks for all the feedback ❤️